### ISSUE TITLE
Autoprefixer generically adds prefixes to browser-specific pseudo-classes

### ISSUE THREAD

**Issue by djibe** [postcss/autoprefixer#1217]
Hi, how to block autoprefixer from adding -webkit prefixes to a Mozilla specific extension ?

Here are some examples to make my point :

```css
.custom-range::-moz-range-thumb {
    -webkit-transition: none;
    transition: none
}
```

`-webkit-transition` has nothing to do here as it is a Mozilla specific selector (`::-moz-range-thumb`).

Another crazy output happens with keyframes and transitions, as Autoprefixer creates useless variants

```css
  -webkit-transition: -webkit-box-shadow 0.1s ease-out;
  transition: -webkit-box-shadow 0.1s ease-out;
  transition: box-shadow 0.1s ease-out;
  transition: box-shadow 0.1s ease-out, -webkit-box-shadow 0.1s ease-out;
```

Instead of just

```css
  -webkit-transition: -webkit-box-shadow 0.1s ease-out;
  transition: box-shadow 0.1s ease-out;
```

How to fix this ? Thanks for your help. And thanks a lot for Autoprefixer !!!

**Comment by ai** [comment]
> how to block autoprefixer from adding -webkit prefixes to a Mozilla specific extension ?

Yeap, it will be nice to avoid webkit prefixes in `:-moz-` rules.

But I will not have time for it in next weeks. Feel free to send PR. You can look how we do it for keyframes

**Comment by ai** [comment]
> -webkit-transition: -webkit-box-shadow 0.1s ease-out;
> transition: box-shadow 0.1s ease-out;

It will not work if `transition` will lost prefix before `box-shadow`

**Comment by djibe** [comment]
Thx a lot.
I'll try to look at the code but I'm not a dev at all.
I'll surely wait ;-)

**Comment by fanich37** [comment]
I'm gonna look at it.

**Comment by ai** [comment]
Released in 9.7.1
